src/user-details/user-details.repository.ts

```diff
@@ -0,0 +1,55 @@
+/* eslint-disable prettier/prettier */
+import {
+  ConflictException,
+  InternalServerErrorException,
+  NotFoundException,
+} from '@nestjs/common';
+import { User } from 'src/auth/user.entity';
+import { EntityRepository, Repository } from 'typeorm';
+import { CreateUserDetailsDto } from './Dto/create-user-details-dto';
+import { UserDetails } from './entity/user-details.entity';
+import { v4 as uuid } from 'uuid';
+import { UpdateUserDetailsDto } from 'src/auth/dto/updateUser-userDetails.dto';
+
+@EntityRepository(UserDetails)
+export class UserDetailsRepository extends Repository<UserDetails> {
+  async getUserDetails(user: User): Promise<UserDetails> {
+    const query = this.createQueryBuilder('userDetails');
+    const { id } = user;
+    try {
+      const details = await query
+        .andWhere('userDetails.userId = :id', {
+          id,
+        })
+        .getOne();
+      return details;
+    } catch (error) {
+      throw new NotFoundException();
+    }
+  }
+
+  async createUserDetailsForUser(
+    user: User,
+    createUserDetailsDto: CreateUserDetailsDto,
+  ): Promise<UserDetails> {
+    const { address, location, number, telephone } = createUserDetailsDto;
+    const userDetails = this.create({
+      address,
+      location,
+      telephone,
+      number,
+      id: uuid(),
```

Comment:
We do not need to define an **id as uuid** because it is already defined in **user-details.entity** ```@PrimaryGeneratedColumn('uuid')```